### PR TITLE
[CARBONDATA-1322] Insert overwrite support and force clean up files and clean up in progress files support added

### PR DESCRIPTION
**Pull request by ravipesala**
The following features are added in this PR.
1. Added support for `LOAD OVERWRITE` and `INSERT OVERWRITE` in carbon load. So after user issues overwrite command all old data will be overwritten with new data.
 Example :
 ```
LOAD DATA INPATH '" data.csv' overwrite INTO table carbontable
```
```
insert overwrite table carbontable select * from othertable
```
When overwrite is in progress no other load will be allowed . And if any other load is already in progress also will be overwritten

2. Added support for force clean table to remove the table with force from disk. It is useful in case of inconsistency with hive metastore. This support is only internal purpose and not exposed to user, so it is supported through scala API not through SQL.

3.  Cleanup the inprogress files while driver is initializing. In case of driver is down while any load is in progress then it must be cleaned while coming up of driver. This is only controlled through parameter `spark.carbon.table.loader.driver` , so it must be set true in driver properties to cleanup the inprogress files.

### CODE REVIEW

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 1]
```diff
@@ -1264,6 +1264,14 @@
 
   public static final String ENABLE_HIVE_SCHEMA_META_STORE_DEFAULT = "false";
 
+  /**
+   * There is more often that in production uses different drivers for load and queries. So in case
+   * of load driver user should set this property to enable loader specific clean up.
+   */
+  public static final String TABLE_LOADER_DRIVER = "spark.carbon.table.loader.driver";
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: I think this property not just for loading, any transactional operation should use this driver. So can you rename it?

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/558/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3151/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala [hunk 2]
```diff
@@ -485,8 +487,8 @@ case class LoadTable(
     }
 
     val dbName = databaseNameOp.getOrElse(sparkSession.catalog.currentDatabase)
-    if (isOverwriteExist) {
-      sys.error(s"Overwrite is not supported for carbon table with $dbName.$tableName")
+    if (isOverwriteTable) {
+      LOGGER.info(s"Overwrite of carbon table with $dbName.$tableName is in progress")
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: should first check whether there is overwrite on going, then do this log

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala [hunk 3]
```diff
@@ -617,4 +621,75 @@ object CommonUtil {
     AttributeReference("partition", StringType, nullable = false,
       new MetadataBuilder().putString("comment", "partitions info").build())()
   )
+
+  def cleanInProgressSegments(storePath: String, sparkContext: SparkContext): Unit = {
```
Comment (jackylk) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: please add some description

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala (hunk 2 above)
Comment (ravipesala) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/carbonTableSchema.scala]: ok

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala (hunk 3 above)
Comment (ravipesala) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: ok

Comment (ravipesala) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: ok

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 1 above)
Comment (ravipesala) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: ok

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3153/

File: integration/spark2/src/main/scala/org/apache/spark/util/CleanFiles.scala [hunk 4]
```diff
@@ -29,12 +29,12 @@ import org.apache.carbondata.api.CarbonStore
 object CleanFiles {
 
   def cleanFiles(spark: SparkSession, dbName: String, tableName: String,
-      storePath: String): Unit = {
+      storePath: String, forceTableClean: Boolean): Unit = {
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/util/CleanFiles.scala]: add default value to `forceTableClean` and add comment for this function

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/560/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3154/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/561/

**Pull request closed without merging**